transformer_engine/pytorch/csrc/extensions.h

```diff
@@ -396,7 +396,8 @@ void nvshmem_finalize();
  * swizzle
  **************************************************************************************************/
 
-void swizzle_scaling_factors(transformer_engine::TensorWrapper &input, bool trans);
+std::optional<at::Tensor> swizzle_scaling_factors(transformer_engine::TensorWrapper &input,
```

Comment:
I don't like this function being in this header, since it is no longer an extension but rather internal utility function.

Reply:
Also I wonder if we shouldn't define a separate type for this return type. For example
```cpp
using BackingStorage = std::optional<std::vector<at::Tensor>>;
```
to make it extra clear what the purpose of this is.

Reply:
Moved it to ``transformer_engine/pytorch/csrc/util.h`.

As for the second comment, I didn't define a seprate type but add comments to this swizzling function and where we hold the tensors.